packages/thirdweb/src/react/core/hooks/transaction/transaction-button-utils.ts

```diff
@@ -113,14 +117,66 @@ export const useTransactionButtonMutation = (
     onError,
     onClick,
   } = props;
+
+  // yep, a state.
+  const [resolvedTransaction, setResolvedTransaction] =
```

Comment:
man i really want to break this api xD

was thinking we could introduce a new prop for this but it might be too confusing / hidden behavior. i guess this is fine for now